core/src/main/java/org/mule/runtime/core/privileged/routing/RoutingResult.java

```diff
@@ -29,18 +33,44 @@ public final class RoutingResult {
 
   private final Map<String, Message> successfulRoutesResultMap;
   private final Map<String, Error> failedRoutesErrorMap;
+  private Map<String, Pair<Error, EventProcessingException>> failedRoutesErrorWithExceptionMap;
 
   public RoutingResult(Map<String, Message> successfulRoutesResultMap, Map<String, Error> failedRoutesErrorMap) {
     this.successfulRoutesResultMap = unmodifiableMap(successfulRoutesResultMap);
     this.failedRoutesErrorMap = unmodifiableMap(failedRoutesErrorMap);
+    this.failedRoutesErrorWithExceptionMap = emptyMap();
+  }
+
+  public static RoutingResult routingResultWithException(Map<String, Message> successfulRoutesResultMap,
+                                                         Map<String, Pair<Error, EventProcessingException>> failedRoutesErrorWithExceptionMap) {
+    RoutingResult routingResult = new RoutingResult(successfulRoutesResultMap, emptyMap());
+    routingResult.setFailedRoutesErrorWithExceptionMap(failedRoutesErrorWithExceptionMap);
+    return routingResult;
+  }
+
+  private void setFailedRoutesErrorWithExceptionMap(Map<String, Pair<Error, EventProcessingException>> failedRoutesErrorWithExceptionMap) {
```

Comment:
Remove this private setter and assing the value using this.failedRoutesErrorWithExceptionMap

Reply:
It was a workaround I used due to `'org.mule.runtime.core.privileged.routing.RoutingResult.this' cannot be referenced from a static context`. Please let me know if it is better to remove this setter and find another workaround.